sljit_src/sljitConfigInternal.h

```diff
@@ -715,6 +720,12 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_sw sljit_exec_offset(void* ptr);
 #define SLJIT_NUMBER_OF_SAVED_REGISTERS 5
 #define SLJIT_LOCALS_OFFSET_BASE 0
 
+#elif (defined SLJIT_CONFIG_S390X && SLJIT_CONFIG_S390X)
+
+#define SLJIT_NUMBER_OF_REGISTERS 12
```

Comment:
I was guessing  that is used for making sure the 2 "reserved" registers are not touched (indeed SLJIT_R0 == r2)

Reply:
Thanks for taking a look at this @carenas and @zherczeg, it's good to see it being worked on again. I am happy to answer questions. Feel free to tag me.

> I was guessing that is used for making sure the 2 "reserved" registers are not touched

Yes, I set the number of registers to 12 because some of the registers are 'reserved' to hold temporary values during codegen.

> same goes for the FPU avoidance

I think we just ignored it in our initial port because (at least at the time) PCRE2JIT didn't use the FPU and we never finished it.

> the use of the constant pool

The constant pool is probably overengineered. You could emit a LLILF-IIHF pair for each constant load. There are definitely some gotchas when modifying instructions like that though. In particular the effect on the performance of the instruction cache can be bad (they usually aren't heavily optimized for self modifying code - I think this is true on x86 too though).

Reply:
Hi @mundaym!
Thank you for joining. I forgot about the stack pointer, but usually 2 temporary registers are enough to implement al instructions. I hope at some point we will only use r0 (since it cannot be used for addressing) and r14 (link register).

On the fly code modification is not frequent, but they often worth it, so their cost should be negligible in the long run.

 I used constant pool on ARMv5, but I decided to never use it again. It make things way too complex.